### PR TITLE
py/py.mk: Add comments explaining how QSTRs are generated.

### PR DESCRIPTION
**Pull request by jimmo**
I thought it might be useful to summarise in one place the process by which `MP_QSTR_FOO` is "magically" transformed into both a QSTR value and an entry in the pool.

Perhaps this belongs in the developer documentation (happy to move it there), but really it's just a description of the various steps in the Makefile. Hopefully almost nobody ever actually needs to know the details of this process, but hopefully it'll save someone some effort one day!

### CODE REVIEW

File: py/mkrules.mk [hunk 1]
```diff
@@ -70,6 +70,7 @@ $(OBJ): | $(HEADER_BUILD)/qstrdefs.generated.h $(HEADER_BUILD)/mpversion.h
 # - if anything in QSTR_GLOBAL_DEPENDENCIES is newer, then process all source files ($^)
 # - else, if list of newer prerequisites ($?) is not empty, then process just these ($?)
 # - else, process all source files ($^) [this covers "make -B" which can set $? to empty]
+# See more information about the QSTR process in py.pk (under "QSTR generation").
```
Comment (stinos) [inline py/mkrules.mk]: py.pk -> py.mk

Also sounds better to say 'For more information about .. se ...' perhaps?

**Comment by dpgeorge** [comment]
I can see why you put it in py.mk, but I would have a weak preference to move it to the developer docs.  At least, there probably should be a qstr section in the developer docs, because it's an often asked question how they work.

**Comment by nevercast** [comment]
By developer docs, do we mean the GitHub Wiki or docs.micropython? 

I usually check docs.micropython for Python related questions, and usually hope that system internals stuff is in the Wiki. Maybe the Wiki predates docs and it should all be migrated ?

**Comment by dpgeorge** [comment]
> By developer docs, do we mean the GitHub Wiki or docs.micropython?

The latter, at: http://docs.micropython.org/en/latest/develop/index.html
This is a relatively new section that it would be good to grow.

> Maybe the Wiki predates docs and it should all be migrated ?

IMO we should disable the github wiki and migrate all its content to either the docs or wiki.micropython.org (the latter itself could do with an update/refresh).

**Comment by jimmo** [comment]
I've moved this over to the developer docs, and added some more info about string interning in general.

**Comment by dpgeorge** [comment]
Thank you, merged in 4ddd46e6cfd1f10efbd6fbbbc0fed520a1058045

**Pull request closed without merging**